### PR TITLE
[SPARK-38450][SQL] Fix HiveQuerySuite//PushFoldableIntoBranchesSuite/TransposeWindowSuite under ANSI mode

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Fix the following test suites under ANSI mode:
* HiveQuerySuite
* HiveTypeCoercionSuite
* HivePartitionFilteringSuite
* PushFoldableIntoBranchesSuite
* TransposeWindowSuite

### Why are the changes needed?

To set up a new GA test job with ANSI mode on

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

Manually turn on ANSI mode and test .
Also it should pass GA tests.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/TransposeWindowSuite.scala [hunk 1]
```diff
@@ -90,14 +90,16 @@ class TransposeWindowSuite extends PlanTest {
   }
 
   test("don't transpose two adjacent windows with intersection of partition and output set") {
-    val query = testRelation
-      .window(Seq(('a + 'b).as('e), sum(c).as('sum_a_2)), partitionSpec3, Seq.empty)
-      .window(Seq(sum(c).as('sum_a_1)), Seq(a, 'e), Seq.empty)
+    if (!conf.ansiEnabled) {
+      val query = testRelation
+        .window(Seq(('a + 'b).as('e), sum(c).as('sum_a_2)), partitionSpec3, Seq.empty)
```
Comment (cloud-fan) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/TransposeWindowSuite.scala:95]: what's wrong with ansi mode here?

**Review by cloud-fan — APPROVED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/TransposeWindowSuite.scala (hunk 1 above)
Comment (gengliangwang) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/TransposeWindowSuite.scala:95]: string + string

**Comment by gengliangwang** [comment]
Btw, this should be the last one. I will double confirm tomorrow.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/TransposeWindowSuite.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/TransposeWindowSuite.scala:95]: I don't think this test case want to test type coercion. Can we add explicit cast or use other string expressions like `concat('a, 'b)`?

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**